tasks/install_system.yml

```diff
@@ -0,0 +1,68 @@
+- name: Make sure iscsi device for {{ vm.host }} @{{ vm.org }} is logged in
+  systemd:
+    name: vm_iscsi@{{ vm.org }}/{{ vm.host }}.service
+    state: started
+  when: vm.storage_type|default(xen_vman_storage_type) == "iscsi"
+
+- name: Format root storage for {{ vm.host }} @{{ vm.org }}
+  shell: mkfs.{{ vm.filesystem|default(xen_vman_default_filesystem) }} "/dev/disk/by-path/ip-{{ xen_vman_iscsi_server }}-iscsi-{{ xen_vman_iscsi_base_wwn }}:{{ vm.org }}-{{ vm.host }}-lun-0"
+  when: vm.storage_type|default(xen_vman_storage_type) == "iscsi"
+
+- name: Make sure vm root is mounted
+  systemd:
+    name: vm_mount@{{ vm.org }}/{{ vm.host }}.service
+    state: started
+
+# TODO: make system installation async
```

Comment:
Shouldn't be async, b/c the other tasks depend on the installed system

Reply:
Maybe @Mr-Pi meant something like you did for the openssh moduli. With that functionality, multiple VMs could be installed in parallel. Also, the executing machine wouldn't have to stay connected.